tests/io/FileFactoryTest.py

```diff
@@ -0,0 +1,46 @@
+import os
+import unittest
+
+from coalib.io.FileFactory import FileFactory
+
+
+class FileFactoryTest(unittest.TestCase):
+
+    def setUp(self):
+        file_factory_test_dir = os.path.join(os.path.split(__file__)[0],
+                                             'FileFactoryTestFiles')
+
+        self.test_file = os.path.join(file_factory_test_dir, 'test.txt')
+        self.uut = FileFactory(self.test_file)
+
+    def test_equal(self):
+        self.assertEqual(self.uut, FileFactory(self.test_file))
+
+    def test_iter(self):
+        self.uut.lines
+        for f in self.uut:
+            self.assertEqual(type(f), str)
+
+    def test_line(self):
+        self.assertEqual(self.uut.get_line(0), 'This is a test file.')
+        self.assertEqual(self.uut.get_line(1), None)
+
+    def test_lines(self):
+        self.assertEqual(self.uut.lines, ('This is a test file.',))
+
+    def test_raw(self):
+        self.assertEqual(self.uut.raw, b'This is a test file.')
+
+    def test_string(self):
+        self.assertEqual(self.uut.string, 'This is a test file.')
+
+    def test_timestamp(self):
+        self.assertEqual(self.uut.timestamp, os.path.getmtime(self.test_file))
+
+    def test_name(self):
+        if os.name == 'posix':
```

Comment:
Hm I dislike such OS-dependent switches.
1. If you already do it, assume that the posix-way is the "right" way ;) Windows' backslashes have been always a pain. So revert the condition: `if os.name == 'windows': ...`
2. can we utilize `os.path` functions to get a representation of the path that is unique across all OSes? Something like invoking `os.path.split` recursively and retrieving a list of path components.
   ```python
   path = get_path_components(self.uut.path)
   self.assertEqual(path, ['tests', 'io', 'FileFactoryTestFiles', 'test.txt'])
   ```